tests/e2e/testsLibrary/che-code/ProjectAndFileTestsCheCode.ts

```diff
@@ -10,24 +10,103 @@
             throw err;
         }
     }
+
+    public async waitAndConfirmTrustAuthorsDialogBox(): Promise<void> {
+        Logger.debug('ProjectAndFileTestsCheCode.waitAndConfirmTrustAuthorsDialogBox');
+        try {
+            Logger.trace('ProjectAndFileTestsCheCode.waitAndConfirmTrustAuthorsDialogBox - waiting for trust authors dialog box overlay');
+            const dialog: WebElement = await this.driverHelper.waitPresence(By.id('monaco-dialog-message-text'));
+            const dialogText: string = await dialog.getText();
+            if (dialogText === ProjectAndFileTestsCheCode.TRUST_AUTHORS_DIALOG_BOX_TEXT) {
+                const trustAuthorsDialogCheckbox: WebElement = await this.driverHelper.waitPresence(By.css('body > div > div.monaco-dialog-modal-block.dimmed > div > div > div.dialog-message-row > div.dialog-message-container > div.dialog-checkbox-row > div.monaco-custom-toggle.codicon.codicon-check.monaco-checkbox'));
+                let isChecked: boolean = await trustAuthorsDialogCheckbox.getAttribute('aria-checked') === 'true';
+                if (!isChecked) {
+                    trustAuthorsDialogCheckbox.click();
+                    isChecked = await trustAuthorsDialogCheckbox.getAttribute('aria-checked') === 'true';
+                    Logger.trace(`Clicked thrust authors dialog checkbox. Is checked: ${isChecked}`);
+                }
+                const trustAuthorsDialogButton: WebElement = await this.driverHelper.waitPresence(By.css('body > div > div.monaco-dialog-modal-block.dimmed > div > div > div.dialog-buttons-row > div > div:nth-child(2) > a'));
+                const trustAuthorsDialogButtonText: string = await trustAuthorsDialogButton.getText();
+                if (trustAuthorsDialogButtonText === ProjectAndFileTestsCheCode.TRUST_AUTHORS_DIALOG_BUTTON_TEXT) {
+                    trustAuthorsDialogButton.click();
+                    Logger.trace(`Clicked trust authors dialog button.`);
+                }
+            }
+        } catch (err) {
+            if (err instanceof error.TimeoutError) {
+                Logger.warn(`ProjectAndFileTestsCheCode.waitAndConfirmTrustAuthorsDialogBox - "Do you trust the authors" dialog not shown. Continuing...`);
+            }
+        }
+    }
+
+    public async waitForRootProjectPresence(workbench: Workbench, sectionTitle: string, rootFolderName: string): Promise<ViewItem> {
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        try {
+            const sideBarView: SideBarView = workbench.getSideBar();
+            const sidebarViewContent: ViewContent = sideBarView.getContent();
+            let sidebarViewSections: ViewSection[];
+            let projectSidebarViewSection: ViewSection;
+            Logger.debug(`ProjectAndFileTestsCheCode.waitForRootProjectPresence - Waiting for ${sectionTitle} section in sidebar:`);
+            while (true) {
```

Comment:
how can we be sure the test will not end up with endless loop?

Reply:
If it's set up correctly, the section with the project name (sectionTitle) will always be present, and if the folder is not found, it will return undefined and instantly fail with and exception after the check.